packages/fxa-content-server/app/scripts/models/account.js

```diff
@@ -455,9 +456,23 @@ const Account = Backbone.Model.extend(
           );
         }
       } catch (err) {
-        // In the event of errors, we don't want to block any user actions, just log it
-        // TODO: Follow up with some better error handling
-        console.log('generateEcosystemAnonId failed...', err);
+        // If we get 412 Precondition Failed it means we
+        // tried to update, but another client beat us to
+        // it. Fetch the new anon ID and use it instead.
+        if (!AuthErrors.is(err, 'ECOSYSTEM_ANON_ID_UPDATE_CONFLICT')) {
+          this.unset('ecosystemAnonId');
+          console.log('generateEcosystemAnonId failed...', err);
+        }
+
+        this.accountProfile().then((profile) => {
```

Comment:
We will need to put `this.accountProfile()` into a try catch so that if it throws an error it doesn't bubble up to user.